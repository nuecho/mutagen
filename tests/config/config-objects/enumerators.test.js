const checkKey = (cfgObject) => {return cfgObject.name === "cfgObjectTest" && cfgObject.tenant === "Environment"};

cfgObjectTests("enumerators", "enumerators-config.json", {initialNumberOfObjects: 23, checkKey: checkKey});
