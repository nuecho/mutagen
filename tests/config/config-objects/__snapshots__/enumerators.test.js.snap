// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen enumerators config import-export] should create the enumerators: the enumerators have been created 1`] = `
Object {
  "stderr": "",
  "stdout": "+ enumerator [Environment/cfgObjectTest]
Completed. 1 object(s) imported.
",
}
`;

exports[`[mutagen enumerators config import-export] should throw exception when new enumerators to import miss mandatory properties enumerators: exception has been thrown 1`] = `
Object {
  "stderr": "Validation failed.
Missing properties:
  - enumerator [Environment/emptyCfgObjectTest]:
    - displayName
    - type
",
  "stdout": "",
}
`;

exports[`[mutagen enumerators config import-export] should update the modified enumerators: the modified enumerators have been updated 1`] = `
Object {
  "stderr": "",
  "stdout": "~ enumerator [Environment/cfgObjectTest]
Completed. 1 object(s) imported.
",
}
`;

exports[`[mutagen enumerators config import-export] should update the same enumerators: the enumerators have been updated 1`] = `
Object {
  "stderr": "",
  "stdout": "~ enumerator [Environment/cfgObjectTest]
Completed. 1 object(s) imported.
",
}
`;
